### DISCUSSION TITLE
Improving the performace of perspective when loading

### DISCUSSION THREAD

**Discussion by harishshivaraj** [perspective-dev/perspective#2273] [Q&A]
Hey all. I have issues with the performance of perspective. I have a data frame with 160k records which I'm trying to load. Now yes that's a lot of data. 

My initial view once loaded is a pivot table of certain configurations which required all the data loaded. 

So my question is, what is the best way to improve the performance? Can I do anything at the perspective level or does it need to be at the data frame level?

I've looked into the streaming option etc, does that mean when I stream chunks of data and the pivot view wouldn't provide the actual results, just because it hasn't got all the data yet? 

Many thanks

**Comment by timkpaine** [comment]
There's not enough information here, please provide a sample repo or code to illustrate the problem. I regularly load more rows than that (though I have no idea how many columns you have in your dataframe).

**Comment by texodus** [comment]
[This example has 320k rows](https://texodus.github.io/nypd-ccrb/). It may take a bit to download the 33mb dataset, but it will load instantly thereafter. We need a lot more information than "it's not fast enough".
